tests/utils_tests.cpp

```diff
@@ -91,4 +91,25 @@ BOOST_AUTO_TEST_CASE(ExtractFilesTest)
   Utils::RemoveFile("./../data/USCensus1990.tar.gz");
 }
 
+/**
+ * Simple test for downloading using curl.
+ */
+BOOST_AUTO_TEST_CASE(CurlDownloadTest)
+{
+  std::string serverName = "https://raw.githubusercontent.com/kartikdutt18/";
```

Comment:
Let's change the source to mlpack's one in the future :)

Reply:
Can we use the path https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/data/test_image.png ?

Reply:
Sure, I'll make the change now.

Reply:
Yes, and the server name will be https://github.com/mlpack/mlpack.git

Reply:
For the curl type download, I am concatenating server name and url to download. Should I just take into input the whole URL and server to distinguish between asio download or curl download.

Reply:
No, I think it's enough :)